dask/dataframe/core.py

```diff
@@ -727,38 +736,41 @@ def apply(func, columns=None):
                               self.frame, self.index)
 
     def sum(self):
-        chunk = lambda df, index: df.groupby(index)[self.key].sum()
-        agg = lambda df: df.groupby(level=0).sum()
+        chunk = lambda df, index: df.groupby(index).sum()
+        agg = lambda df: df.groupby(self.index).sum()
         return aca([self.frame, self.index],
                    chunk=chunk, aggregate=agg, columns=[self.key])
 
     def min(self):
-        chunk = lambda df, index: df.groupby(index)[self.key].min()
-        agg = lambda df: df.groupby(level=0).min()
+        chunk = lambda df, index: df.groupby(index).min()
+        agg = lambda df: df.groupby(self.index).min()
         return aca([self.frame, self.index],
                    chunk=chunk, aggregate=agg, columns=[self.key])
 
     def max(self):
-        chunk = lambda df, index: df.groupby(index)[self.key].max()
-        agg = lambda df: df.groupby(level=0).max()
+        chunk = lambda df, index: df.groupby(index).max()
+        agg = lambda df: df.groupby(self.index).max()
```

Comment:
> Why these changes?

The easiest way to implement `resample` in dask is to use the `pd.TimeGrouper` object. Currently, only a column is allowed to be the grouper, but to allow us to use `pd.TimeGrouper` we need to allow other kinds of objects. This is a WIP, but I wanted to get feedback on the approach before going further.